### PR TITLE
Fix memory leak in calibration

### PR DESCRIPTION
**Pull request by peressounko**

### CODE REVIEW

**Comment by peressounko** [comment]
Dear @davidrohr, could you please have a look at failed tests? I am not sure if double defined "configKeyValues" is related to PHOS?

**Comment by peressounko** [comment]
Dear @davidrohr, could you please have a look at failed tests? I am not sure if double defined "configKeyValues" is related to PHOS?

**Comment by davidrohr** [comment]
Are they still happening. I thought I fixed them 2 days ago. Will check tonight.

Kind Regards
David Rohr

Sent from my mobile. (Excuse the typos!)

**Comment by davidrohr** [comment]
@peressounko : The problem with the configKeyValues was already fixed. The FullCI for this PR is failing with:
```
[20704:BadMapCalibSpec]: [ERROR] error parsing options of o2-phos-calib-workflow: option '--meta-output-dir' is ambiguous and matches different versions of '--meta-output-dir'
```
Which seems to be related to PHOS calib.

**Comment by peressounko** [comment]
Dear @martenole, could you please have a look at this error? 
[20704:BadMapCalibSpec]: [ERROR] error parsing options of o2-phos-calib-workflow: option '--meta-output-dir' is ambiguous and matches different versions of '--meta-output-dir'
I defined meta-output-dir option in one place (phos-calib-workflow.cxx) and do not see how can it interfere with something else? Can it interfere with other frameworks?
     Thanks and best regards, Dmitri

**Comment by davidrohr** [comment]
@ktf : could you have a look what goes wrong here? At first glance, I see that they have indeed specified it only once, and as far as I can tell it should not collide with any global option. But don't have much time right now to dig deeper.

**Comment by ktf** [comment]
meta-output-dir is definitely not a framework option. I will check.

File: Detectors/PHOS/calib/src/phos-calib-workflow.cxx [hunk 1]
```diff
@@ -33,6 +33,10 @@ void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
   workflowOptions.push_back(ConfigParamSpec{"energy", o2::framework::VariantType::Bool, false, {"collect tree for E calib"}});
   workflowOptions.push_back(ConfigParamSpec{"badmap", o2::framework::VariantType::Bool, false, {"do bad map calculation"}});
 
+  workflowOptions.push_back(ConfigParamSpec{"output-dir", o2::framework::VariantType::String, "./", {"ROOT files output directory"}});
+  workflowOptions.push_back(ConfigParamSpec{"meta-output-dir", o2::framework::VariantType::String, "./", {"metafile output directory"}});
```
Comment (ktf) [inline Detectors/PHOS/calib/src/phos-calib-workflow.cxx]:
```suggestion
  workflowOptions.push_back(ConfigParamSpec{"meta-output-dir", o2::framework::VariantType::String, "/dev/null", {"metafile output directory"}});
```

other workflows also define it so IIRC they need to be uniform in their default value.

**Comment by ktf** [comment]
I think it's because we have meta-output-dir defined by multiple workflows and they have different default values.

**Comment by peressounko** [comment]
Dear @ktf, could you please have a look once more? Even after your fix message 
[9027:BadMapCalibSpec]: [ERROR] error parsing options of o2-phos-calib-workflow: option '--meta-output-dir' is ambiguous and matches different versions of '--meta-output-dir'
still present. Thanks, Dmitri

**Comment by ktf** [comment]
i will check

**Comment by peressounko** [comment]
Dear @ktf, should I remove modifications of options and leave memory leak fix for this PR? I can not commit it more than a week
    Best regards, Dmitri

**Comment by davidrohr** [comment]
@peressounko : Yes, please, fix only the leak for now and move the change of the options to a separate PR for now. Sorry for the inconvenience.

**Comment by ktf** [comment]
Yes, sorry, this slipped through.

**Comment by peressounko** [comment]
Dear @davidrohr, could you please have a look at current state of this PR? It failed in fillCI, but not it looks unrelated to PHOS. 
To avoid problems with --metafile or --output-dir options for the moment I changed option names.
     Best regards, Dmitri

**Pull request merged**